File: internal/bot/command/delete_group.go
package command

import (
	"strconv"

	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api/v5"
	"github.com/miti99/store-scraper-bot-go/internal/config"
	"github.com/miti99/store-scraper-bot-go/internal/repository"
)

// /delgroup [groupId] — Java DeleteGroupCommand. Admin-only.
type DeleteGroupCommand struct {
	cfg       *config.Config
	adminRepo *repository.AdminRepository
	groupRepo *repository.GroupRepository
}

func NewDeleteGroupCommand(cfg *config.Config, adminRepo *repository.AdminRepository, groupRepo *repository.GroupRepository) *DeleteGroupCommand {
	return &DeleteGroupCommand{cfg: cfg, adminRepo: adminRepo, groupRepo: groupRepo}
}

func (c *DeleteGroupCommand) Execute(msg *tgbotapi.Message, sender Sender) {
	if !requireAdminUser(msg.From.ID, msg.Chat.ID, c.cfg, sender) {
		return
	}
	args := splitArgs(msg.CommandArguments())
	if len(args) > 1 {
		_ = sender.SendMessage(msg.Chat.ID, "Invalid arguments")
		return
	}
	groupID := msg.Chat.ID
	if len(args) == 1 {
		parsed, err := strconv.ParseInt(args[0], 10, 64)
		if err != nil {
			_ = sender.SendMessage(msg.Chat.ID, "Invalid arguments")
			return
		}
		groupID = parsed
	}
	removed, err := c.adminRepo.RemoveGroup(groupID)
	if err != nil {
		_ = sender.SendMessage(msg.Chat.ID, "Internal server error")
		return
	}
	if !removed {
		_ = sender.SendMessage(msg.Chat.ID, "Group is not added")
		return
	}
	_ = sender.SendMessage(msg.Chat.ID, "Group deleted successfully")
}

File: src/bot/commands/delete-group.js
import { requireAdminUser } from './command-utils.js';

// /delgroup [groupId] — admin-only. Removes from allowlist AND wipes the
// group's tracked-app state to avoid orphaned `group:{chatId}` keys.
export function createDeleteGroupCommand(config, store) {
  return async (msg, sender, args) => {
    if (!(await requireAdminUser(msg.from.id, msg.chat.id, config, sender))) return;
    if (args.length > 1) {
      await sender.sendMessage(msg.chat.id, 'Invalid arguments');
      return;
    }
    let groupId = msg.chat.id;
    if (args.length === 1) {
      const parsed = Number.parseInt(args[0], 10);
      if (!Number.isFinite(parsed)) {
        await sender.sendMessage(msg.chat.id, 'Invalid arguments');
        return;
      }
      groupId = parsed;
    }
    const removed = await store.admin.removeGroup(groupId);
    if (!removed) {
      await sender.sendMessage(msg.chat.id, 'Group is not added');
      return;
    }
    await store.group.deleteGroup(groupId);
    await sender.sendMessage(msg.chat.id, 'Group deleted successfully');
  };
}
